Mathlib/Combinatorics/Quiver/Path/Vertices.lean

```diff
@@ -0,0 +1,236 @@
+  | cons p' e ih =>
+    simp only [vertices_cons, length_cons, List.length_concat, ih]
+
+lemma length_vertices_pos {a b : V} (p : Path a b) :
+    0 < p.vertices.length := by simp
+
+lemma vertices_nonempty {a : V} {b : V} (p : Path a b) : p.vertices ≠ [] := by
+  rw [← List.length_pos_iff_ne_nil, vertices_length]; omega
+
+/-- The head of the vertices list is the start vertex -/
+@[simp]
+lemma vertices_head? {a b : V} (p : Path a b) : p.vertices.head? = some a := by
+  induction p with
+  | nil => simp only [vertices_nil, List.head?_cons]
+  | cons p' e ih =>
+    simp only [vertices_cons]
+    have : ¬p'.vertices.isEmpty := by
+      simp_all only [List.isEmpty_iff]
+      apply Aesop.BuiltinRules.not_intro
+      intro a_1
+      simp_all only [List.head?_nil, reduceCtorEq]
+    simp
+    simp_all only [List.isEmpty_iff, Option.getD_some]
+
+@[simp]
+lemma length_toPath {a b : V} (e : a ⟶ b) : (e.toPath).length = 1 := rfl
+
+@[simp]
+lemma vertices_getLast {a b : V} (p : Path a b) (h : p.vertices ≠ []) :
+  p.vertices.getLast h = b := by
+  induction p with
+  | nil => simp only [vertices_nil, List.getLast_singleton]
+  | cons p' e ih =>
+    simp only [vertices_cons]
+    rw [@List.getLast_concat']
+
+@[simp]
+lemma vertices_comp {a b c : V} (p : Path a b) (q : Path b c) :
+  (p.comp q).vertices = p.vertices.dropLast ++ q.vertices := by
+  induction q with
+  | nil =>
+    simp only [comp_nil, vertices_nil]
+    have h_nonempty : p.vertices.length > 0 := by simp
+    have h_ne_nil : p.vertices ≠ [] := List.ne_nil_of_length_pos h_nonempty
+    rw [← List.dropLast_append_getLast h_ne_nil, vertices_getLast p h_ne_nil]
+    simp_all only [gt_iff_lt, ne_eq, List.cons_ne_self, not_false_eq_true,
+      List.dropLast_append_of_ne_nil, List.dropLast_singleton, List.append_nil]
+  | cons q' e ih =>
+    simp only [comp_cons, vertices_cons, ih, List.concat_eq_append]
+    rw [List.append_assoc]
+
+lemma start_mem_vertices {a b : V} (p : Path a b) : a ∈ p.vertices := by
+  induction p with
+  | nil => simp only [vertices_nil, List.mem_cons, List.not_mem_nil, or_false]
+  | cons p' e ih =>
+    simp only [vertices_cons]
+    simp_all only [List.concat_eq_append, List.mem_append, List.mem_cons, List.not_mem_nil,
+      or_false, true_or]
+
+@[simp] lemma length_eq_zero_iff {a : V} (p : Path a a) :
+    p.length = 0 ↔ p = Path.nil := by
+  cases p
+  · simp only [length_nil]
+  · constructor
+    · intro h; cases h
+    · intro h; cases h
+
+/-- Given vertices lists from a path composition, the prefix path's vertices is a prefix of
+the full path's vertices -/
+lemma isPrefix_dropLast_of_comp_eq {V : Type*}
+    [Quiver V] {a b c : V} {p : Path a b} {p₁ : Path a c} {p₂ : Path c b}
+    (h : p.vertices = p₁.vertices.dropLast ++ p₂.vertices) :
+    p₁.vertices.dropLast.IsPrefix p.vertices := by
+  rw [h]
+  exact List.prefix_append p₁.vertices.dropLast p₂.vertices
+
```

Comment:
I think this is too specialised / too thin a wrapper around `List.prefix_append` to be worth having.
```suggestion
```